src/interfaces/records-write.ts

```diff
@@ -399,7 +400,7 @@ export class RecordsWrite {
       await ProtocolAuthorization.authorize(tenant, this, this, messageStore);
     } else if (this.author === tenant) {
       // if author is the same as the target tenant, we can directly grant access
-    } else if (this.author !== undefined && this.authorizationPayload?.permissionsGrantId !== undefined) {
+    } else if (this.author !== undefined && this.authorizationPayload!.permissionsGrantId !== undefined) {
```

Comment:
Correct me if I'm wrong, but I think if `author` is present, then `authorizationPayload` will be present. Or maybe your currently open PR changes things.

Reply:
You are right that `authorizationPayload` should be present.